ci-operator/step-registry/cucushift/winc/prepare/cucushift-winc-prepare-commands.sh

```
@@ -297,16 +297,94 @@ echo "Windows OS image ID: $windows_os_image_id"
 
+echo "DEBUG: Public key derived from cluster profile ssh-privatekey:"
+ssh-keygen -y -f "${CLUSTER_PROFILE_DIR}/ssh-privatekey" || echo "Failed to derive public key"
+
 echo "DEBUG: Windows machineset configuration:"
 oc get machineset -n openshift-machine-api ${winworker_machineset_name} -o yaml
 
+echo "DEBUG: Initial machine objects:"
+oc get machines -n openshift-machine-api -o wide | grep -E "NAME|winworker" || echo "No Windows machines found yet"
+for machine in $(oc get machines -n openshift-machine-api -o name | grep winworker); do
+  echo "=== ${machine} ==="
+  oc get -n openshift-machine-api ${machine} -o yaml
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)